src/relay/transforms/merge_composite.cc

```diff
@@ -36,22 +36,24 @@ namespace tvm {
 namespace relay {
 namespace merge_composite {
 
-Function InferType(const Function& expr) {
-  auto mod = IRModule::FromExpr(expr);
+Function InferType(const Function& expr, const IRModule& m) {
+  IRModule mod(m);
+  mod->Update(mod->GetGlobalVar("main"), expr);
```

Comment:
Since now we update the original module with new transformed function, should we update the corresponding function instead of `main`?

Reply:
mmm, you're right, reading the pass infrastructure a little more today. 

The FunctionPass, however, doesn't seem to pass the information on what Function we're see down to the passes: https://github.com/apache/incubator-tvm/blob/8578096853eec5711bfcc9a3a68145fd12a135cb/src/relay/ir/transform.cc#L123-L132

I guess we can either change that API (which touches a lot of passes), or maybe invert this Map https://github.com/apache/incubator-tvm/blob/4347b41a5e64a2a453297b371232d6e101051b3c/include/tvm/ir/module.h#L53, find the global var, and store that in the class.

Reply:
Or maybe we can rewrite MergeComposite to be a module pass so that we can iterate `functions` by ourselves.

Reply:
> Since now we update the original module with new transformed function, should we update the corresponding function instead of `main`?

For now, other functions in module don't call 'main', so it is safe to replace 'main'. If we are infering function which is mutated from 'main', that's just what we want, if we are infering other functions, it will be a duplicated function but it doesn't harm.  And, with only a mutated function, it seems we can't find it in original global_var_map_ because we don't know its crossponding global variable's name.

Reply:
Make sense, although I still prefer to avoid dummy modules. I think it's fine to let his PR in first, and maybe I can fix that later on.